src/atomic_file/atomic.rs

```diff
@@ -309,10 +310,58 @@ mod tests {
         let version_2_path = file.path(2);
         let rename_path =
             root.join(format!("{}_cellphoneId.1", root.display()));
-        std::fs::rename(version_2_path, rename_path).unwrap();
+        fs::rename(version_2_path, rename_path).unwrap();
         // We should take content from current machine
         let current = file.load().unwrap();
         let content = current.read_to_string().unwrap();
         assert_eq!(content, current_machine);
     }
+
+    #[rstest]
+    #[case(3, &[1, 3], "case_1")]
+    #[case(5, &[2, 4], "case_2")]
+    #[case(10, &[3, 5, 7, 9, 10], "case_3")]
+    #[case(15, &[5, 14, 15], "case_4")]
+    fn latest_version(
+        #[case] versions: usize,
+        #[case] cellphone_versions: &[usize],
+        #[case] temp_name: &str,
+    ) {
+        // Create the files without atmic to handles files names
+        // let versions = 10;
+        // let cellphone_versions = &[3, 5, 7, 9, 10];
+        let dir = TempDir::new(temp_name).unwrap();
+        let root = dir.path();
+        // let root = PathBuf::from("./latest_version");
+        let file = AtomicFile::new(&root).unwrap();
```

Comment:
Can we move this line closer to its usage? I.e. right before `assert_eq!(file.latest_version().unwrap().1, versions)`

Reply:
It is needed to get the prefix of the file. But I defined it just before its usage